File: ComputeDecimalRepresentation.py
class Solution:
    def decimalRepresentation(self, n: int) -> list[int]:
        ans = []
        permuta = 0

        while n > 0:
            if n % 10 > 0:
                ans.append((n % 10) * (10**permuta))
            n //= 10
            permuta += 1
        ans.reverse()   
        return ans

solutionTest = Solution()

valueTest = 537

print(solutionTest.decimalRepresentation(valueTest))

File: ComputeDecimalRepresentation.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:

    vector<int> decimalRepresentation(int n) {
        vector<int> ans;
        int permuta = 0;
        int fator = 1;

        while(n > 0){
            int digit = n % 10;

            if(digit > 0){
                ans.push_back(digit * fator);

            }
            n/=10;
            fator*=10;
        }

        reverse(ans.begin(), ans.end());
        return ans;

    }


};

int main(){
    Solution solutionTest;

    int valueTest = 537;
    vector<int> result = solutionTest.decimalRepresentation(valueTest);

    cout << "[ ";
    for (int num : result) {
        cout << num << " ";
    }
    cout << "]" << endl;

    return 0;
}
